src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -282,16 +304,61 @@ public void WriteSchemaObjectExampleValue(CodeWriter writer, ObjectType sot, Exa
 
             if (hasUnconsumedProperties)
                 WriteSchemaObjectExampleProperties(writer, sot, ev, variableName, consumedProperties);
+
+            // assign readonly list and dictionary properties.
+            foreach (var objectType in sot.EnumerateHierarchy())
+            {
+                foreach (var targetProperty in objectType.Properties)
+                {
+                    if (consumedProperties.Contains(targetProperty))
+                        continue;
+                    var paramValue = FindPropertyValue(sot, ev, targetProperty!);
+                    if (paramValue is not null)
+                    {
+                        assignmentWriterDelegates.Enqueue(CreateAssignmentWriterDelegate(targetProperty.ValueType, paramValue, $"{PropertyVaraibleName(variableName, targetProperty)}"));
+                    }
+                }
+            }
         }
 
-        protected void CreateTagWriterDelegate(FormattableString newVariableName, CSharpType valueType, ExampleValue ev)
+        public CodeWriterDelegate CreateAssignmentWriterDelegate(CSharpType cst, ExampleValue exampleValue, FormattableString variableName)
         {
-            _tagsWriterDelegate = new CodeWriterDelegate(writer =>
+            return new CodeWriterDelegate(writer =>
+            {
+                if (cst.Name == "IList" || cst.Name == "IEnumerable")
+                {
+                    writer.UseNamespace("System.Collections.Generic");
+                    var idx = 0;
+                    foreach (var element in exampleValue.Elements ?? new List<ExampleValue>())
```

Comment:
Here exampleValue.Elements is come from testmodeler, and may be `null` since incorrect example file.

I will add  default value in codeModelPartial like below to avoid ?? check in generation code.
~~~
[YamlMember(Alias = "elements")]
public System.Collections.Generic.ICollection<ExampleValue> Elements { get; set; } = Array.Empty<ExampleValue>();
~~~